### ISSUE TITLE
Axis title could also change the labels in the hovering popup

### ISSUE THREAD

**Issue by adube** [plotly/plotly.js#2618]
This is an enhancement suggestion.

When setting the title of an axis, such as:

```
        xaxis: {
            type: 'linear',
            zeroline: false,
            title: "hellox"  // <----
        },
```

it would make sense to have the same title as default value in the popup being shown on mouse hovering.  Currently, the default value stays "x", where it could have been "hellox" instead.

See an example here: https://codepen.io/anon/pen/qoGQvx

See also the riginal question asked on [Stack Overflow](https://stackoverflow.com/questions/49820560/how-to-change-only-the-title-of-hovering-popup-in-plotly-js?noredirect=1#comment87407776_49820560)

Please, let me know if that makes sense.

**Comment by alexcjohnson** [comment]
Thanks @adube - yes, makes a lot of sense. I don't think we can *always* do this, there are times when the title is too long, or displayed on one subplot and not others, etc... so it would probably have to be opt-in unless we can come up with a sufficiently robust criterion for automatically using the title.

**Comment by adube** [comment]
@alexcjohnson Thanks for your comment.

I agree.  As a way to not _always_ enable this behaviour, we could have some sort of "flag" option (boolean) which would be `false` by default to not break the current behaviour.  When setting it to `true`, users could expect limitations such as title being too long and the other things you mentioned.

I'm not familiar enough with the code to be able to determine a more robust solution at this point, but I think the above suggestion could do the trick.

**Comment by nicolaskruchten** [comment]
I think this makes a ton of sense in general and especially in 3d and ternary contexts: 3d because after a few rotations or whatever, it's no longer visually clear what `x`, `y`, and `z` mean, and ternary because there's no reasonable way a user could know what `A`, `B` and `C` map to ;)

**Comment by etpinard** [comment]
Somewhat a duplicate of https://github.com/plotly/plotly.js/issues/265

**Comment by adube** [comment]
Indeed, it is.

**Comment by vvch** [comment]
> I agree. As a way to not _always_ enable this behaviour, we could have some sort of "flag" option (boolean) which would be `false` by default to not break the current behaviour. When setting it to `true`, users could expect limitations such as title being too long

This option (e.g., 'hovertitle') may also contain text (for each axis) as alternative short hover-only title for this axis. That is, default 'false' value means current behaviour, 'true' means use axis label (maybe with some automatic abbreviation), and text value means use this value.

**Comment by vvch** [comment]
Seems like similar feature was already implemented (but not pulled yet): #3126 #3007

**Comment by nicolaskruchten** [comment]
@antoinerg this would be a nice addition :)
